### DISCUSSION TITLE
Improve subnet consensus (22Q2 brainstorming)

### DISCUSSION THREAD

**Discussion by jsoares** [consensus-shipyard/consensuslab#135] [Project proposals]
* Optimize performance / scalability
* Advanced tooling for simulation / testing / debugging / benchmarking / deploying
* Extend selection of protocols (Asynchronous consensus for subnets (maybe an RFP))
* Common-coins and Asynchronous BFT (for DDoS resilience and fairness
* Exploring alternative approaches to implementing consensus/distributed systems, learning from experience with Mir
    * [Pull based protocols](https://github.com/protocol/ConsensusLab/discussions/176)
    * Protocol logic code expressed in "Control-oriented" notation
    * Paper
* General approach for modular and efficient reconfiguration, durability, checkpoints, crash recovery etc.
* Add consensus protocols into Mir (e.g. Tendermint in Mir)

@matejpavlovic @quema @dnkolegov @sergefdrv @vukolic @sa8 @guy-goren @LefKok

**Comment by adlrocha** [comment]
Related to this, proof-of-consensus: https://blog.succinct.xyz/post/2022/09/20/proof-of-consensus

It is strongly related with something we've been discussing when brainstorming about the integration of Lurk in IPC. Mainly, the ability to include succinct proofs of the consensus verification in every block of a subnet. Having this as a default feature for IPC subnets can improve the checkpointing protocol and relax the requirement of full-nodes of having to sync with all its ancestors.
